newIDE/app/src/EventsSheet/InstructionEditor/InstructionOrObjectSelector.js

```diff
@@ -145,8 +169,77 @@ export default class InstructionOrObjectSelector extends React.PureComponent<
     if (this._selectedItem.current && this._scrollView.current) {
       this._scrollView.current.scrollTo(this._selectedItem.current);
     }
+    const { allObjectsList, allGroupsList } = enumerateObjectsAndGroups(
+      this.props.globalObjectsContainer,
+      this.props.objectsContainer
+    );
+
+    this.instructionSearchApi = new Fuse(
+      deduplicateInstructionsList(this.allInstructionsInfo),
+      {
+        ...sharedFuseConfiguration,
+        keys: [
+          { name: 'displayedName', weight: 2 },
+          { name: 'fullGroupName', weight: 1 },
+        ],
+      }
+    );
+    this.objectSearchApi = new Fuse(allObjectsList, {
+      ...sharedFuseConfiguration,
+      getFn: (item, property) => item.object.getName(),
+      keys: ['name'], // Not used as we only use the name of the object
+    });
+    this.groupSearchApi = new Fuse(allGroupsList, {
+      ...sharedFuseConfiguration,
+      getFn: (item, property) => item.group.getName(),
+      keys: ['name'], // Not used as we only use the name of the group
+    });
+    this.tagSearchApi = new Fuse(this._getAllObjectTags(), {
+      ...sharedFuseConfiguration,
+    });
   }
 
+  _search = (searchText: string) => {
+    if (searchText === '') return;
+
+    this.setState({
+      searchResults: {
+        objects: this.objectSearchApi
+          ? this.objectSearchApi.search(`'${searchText}`).map(result => ({
```

Comment:
What is this ' useful for?

Reply:
I'm not sure but the results felt better : https://fusejs.io/examples.html#extended-search